modules/openthread/platform/hdlc_interface.hpp

```diff
@@ -0,0 +1,206 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
+ *  Copyright (c) 2022-2024, NXP.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
```

Comment:
@dleach02 I see it has 3-clause BSD license now. Is this allowed for new Zephyr contributions? Does it need TSC approval?

Reply:
I think you do need to get TSC approval. This code looks like it started from the OpenThread Authors so we can't change the license.

Reply:
@rlubos the following seems to be the process we may need to follow: https://docs.zephyrproject.org/latest/contribute/external.html#external-src-process

I'll fill out the RFC request to present to the TSC.

Reply:
#79882

@rlubos @JA-NXP   Please comment/edit the RFC directly to address any problems in my wording